packages/blade/src/components/Input/BaseInput/getBaseInputStyles.ts

```diff
@@ -1,26 +1,74 @@
 import type { CSSObject } from 'styled-components';
+import type { BaseInputProps } from './BaseInput';
 import type { Theme } from '~components/BladeProvider';
 import getTextStyles from '~components/Typography/Text/getTextStyles';
 import { makeBorderSize, makeSpace } from '~utils';
 
-const getBaseInputStyles = ({ theme }: { theme: Theme }): CSSObject => ({
+export const getInputBackgroundAndBorderStyles = ({
+  theme,
+  isFocused,
+  isDisabled,
+  validationState,
+}: Pick<BaseInputProps, 'isDisabled' | 'validationState'> & {
+  isFocused?: boolean;
+  theme: Theme;
+}): CSSObject => {
```

Comment:
Let's not inline complex types on function body, looks very cluttered. 

Also `getInputBackgroundAndBorderStyles` & `getBaseInputStyles` has the exact same type defs, check if we can just reuse it.

```ts
type GetBaseInputStylesProps = Pick<BaseInputProps, 'isDisabled' | 'validationState'> & {
  isFocused?: boolean;
  theme: Theme;
};
```